sc/social/like/vocabularies.py

```diff
@@ -46,3 +54,18 @@ def __call__(self, context):
     SimpleTerm(value=u'Like', title=_(u'Like')),
     SimpleTerm(value=u'Share', title=_(u'Share')),
 ])
+
+
+@provider(IVocabularyFactory)
+def ImageScalesVocabulary(object):
+    """Obtains available scales from plone.app.imaging"""
+    terms = []
+    for scale, (width, height) in getAllowedSizes().iteritems():
+        if width >= LEAD_IMAGE_MIN_WIDTH and height >= LEAD_IMAGE_MIN_HEIGHT:
+            translated = _(
+                'imagescale_{0:s}'.format(scale),
+                default='{0:s} ${{width}}x${{height}}'.format(scale),
+                mapping={'width': str(width), 'height': str(height)})
+            terms.append(SimpleTerm(scale, scale, translated))
+
+    return SimpleVocabulary(terms)
```

Comment:
how about to filter the result of the default vocabulary?
 https://github.com/plone/plone.app.vocabularies/blob/2.1.x/plone/app/vocabularies/images.py#L13

Reply:
I do not think so.

> ScalesVocabulary(context)._terms[0].title
> u'imagescale_mini'

> ScalesVocabulary(context)._terms[0].value
> 'mini'

> ScalesVocabulary(context)._terms[0].value
> 'token'

Reply:
Test #130 #pullrequestreview-62926659

Reply:
why not? that way we can skip the i18n.